diff --git a/staticgen.go b/staticgen.go
index 63759e8..5a0b2a0 100644
--- a/staticgen.go
+++ b/staticgen.go
@@ -70,15 +70,9 @@ func (g *Generator) Run(ctx context.Context) error {

 // Start loads configuration from ./static.json, starts the
 // configured server, and begins the crawling process.
-func (g *Generator) Start(ctx context.Context) error {
-	// load configuration
-	err := g.Config.Load("static.json")
-	if err != nil {
-		return fmt.Errorf("loading configuration: %w", err)
-	}
-
+func (g *Generator) Start(ctx context.Context, config Config) error {
 	// remove output dir
-	err = os.RemoveAll(g.Dir)
+	err := os.RemoveAll(g.Dir)
 	if err != nil {
 		return fmt.Errorf("removing output directory: %w", err)
 	}
@@ -49,8 +49,8 @@ type Generator struct {

 // Run starts the configured server command, starts to perform crawling,
 // and waits for completion before shutting down the configured server.
-func (g *Generator) Run(ctx context.Context) error {
-	err := g.Start(ctx)
+func (g *Generator) Run(ctx context.Context, config Config) error {
+	err := g.Start(ctx, config)
 	if err != nil {
 		return fmt.Errorf("starting: %w", err)
 	}
